controller/eks-cluster-config-handler.go

```diff
@@ -109,13 +109,17 @@ func (h *Handler) recordError(onChange func(key string, config *eksv1.EKSCluster
 		}
 		if err != nil {
 			if !strings.Contains(err.Error(), "currently has update") {
-				// the update is valid in that the controller should retry but there is
-				// no actionable resolution as far as a user is concerned. An update
-				// that has either been initiated by eks-operator or another source is
-				// already in progress. It is possible an update is not being immediately
-				// reflected in upstream cluster state. The config object will reenter the
-				// controller and then the controller will wait for the update to finish.
+				// The update is valid in that the controller should retry but there is no actionable resolution as far
+				// as a user is concerned. An update that has neither been initiated by the eks-operator or another
+				// source is in already in progress. It is possible an update is not being immediately reflected in
+				// upstream cluster state. The config object will reenter the controller and then the controller will
+				// wait for the update to finish.
 				message = err.Error()
+
+				// If we have hit a retriable error, we don't know whether that error will resolve or not. Wait in this
+				// case to avoid spamming the EKS logs and thrashing other updates.
+				logrus.Infof("waiting for cluster [%s] update to complete", config.Name)
+				time.Sleep(30 * time.Second)
```

Comment:
I don't think we want to block the thread like this. I think it's actually returning an error which reenqueues the controller since the status is a subresource iirc, so changes to it do cause a reenqueue. Reenqueueing on errors is the backboen to the framework and shouldn't be skipped like this. I would also think that we want the first occurrence to log immediately. There is a builtin backoff that should slow it down over time iirc. If you can show us an example of the offending error and we may be able to think of a different solution.